### PR TITLE
chore!: rename `GatewayBot` to `GatewayClient`

### PR DESCRIPTION
**Pull request by EnokiUN**
This pull request addresses the result of a conversation we had, basically we've concluded that the library offers a client while the thing you make with it *may* be a bot.

This pull request changes how velum references to the client as bots by referencing to them as clients

### CODE REVIEW

File: velum/impl/bot.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 _LOGGER = logging.getLogger("velum.bot")
```
Comment (Sharp-Eyes) [inline velum/impl/bot.py]: This should be renamed to `"velum.client"`

**Review by Sharp-Eyes — CHANGES_REQUESTED** [review]
We should probably then also
- rename `velum/impl/bot.py` to `velum/impl/client.py`
- refactor the examples to use `GatewayClient`
- refactor the short example in the readme to use `GatewayClient`
- refactor docstrings to use client instead of bot (mostly events)

File: velum/impl/bot.py (hunk 1 above)
Comment (ooliver1) [reply velum/impl/bot.py]: :clown_face:

**Review by Sharp-Eyes — APPROVED** [review]
lgtm!

**Pull request merged**